app/models/concerns/issues/grouped_export.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+module Issues::GroupedExport
+  extend ActiveSupport::Concern
+
+  CSV_OPTIONS       = { col_sep: ';' }
+  HEADERS_SEPARATOR = '⁻-⁻'
+
+  module ClassMethods
+    def grouped_export
+      csvs_by_script = {}
+      grouped_ids    = ids_grouped_by_script_id
+      scripts        = Script.where(id: grouped_ids.keys).pluck(:id, :name).to_h
+
+      grouped_ids.each do |script_id, issue_ids|
+        acc         = {}
+        script_name = scripts[script_id]
+
+        summarized_find_each_for_ids issue_ids do |issue|
+          issue.grouped_export script_name, acc
+        end
+
+        grouped_csvs = build_csvs_for_groups acc
+        filename     = sanitize_filename(script_name) + '.zip'
+
+        csvs_by_script[filename] = tmp_zip_file_content grouped_csvs
+      end
+
+      hash_to_zip csvs_by_script
+    end
+
+    def summarized_find_each_for_ids ids
+      Issue.where(id: ids).select(:id, :description, :data).find_each(batch_size: 100) do |issue|
+        yield issue
+      end
+    end
+
+    def ids_grouped_by_script_id
+      scripts_with_issues = {}
+      select_columns      = ["#{Issue.table_name}.id", "#{Job.table_name}.script_id"]
+
+      all.joins(run: :job).pluck(*select_columns).each do |issue_id, script_id|
+        scripts_with_issues[script_id] ||= []
+        scripts_with_issues[script_id]  << issue_id
+      end
+
+      scripts_with_issues
+    end
+
+    def build_csvs_for_groups groups
+      require 'csv'
+
+      csvs = {}
+
+      groups.each do |name_with_headers, rows|
+        name, *headers = name_with_headers.split HEADERS_SEPARATOR
+        filename       = grouped_sanitized_filename name, csvs
+
+        csvs[filename] = CSV.generate CSV_OPTIONS do |csv|
+          csv << headers
+
+          rows.each { |row| csv << row}
+        end
+      end
+
+      csvs
+    end
+
+    def sanitize_filename filename
+      splited_name = filename[0, 200].split /(?<=.)\.(?=[^.])(?!.*\.[^.])/m
+
+      splited_name.map { |s| s.gsub /[^a-z0-9\-\_\.]+/i, '_' }.join '.'
+    end
+
+    def grouped_sanitized_filename name, csvs
+      filename = sanitize_filename(name) + '.csv'
```

Comment:
Por lo mismo de más arriba:
```suggestion
      filename = "#{sanitize_filename name}.csv"
```